addons/mass_mailing/models/mailing_contact.py

```diff
@@ -34,11 +42,23 @@ def create(self, vals):
         return super(MassMailingContactListRel, self).create(vals)
 
     def write(self, vals):
-        if 'opt_out' in vals:
-            vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now()
+        for record in self:
+            vals['opt_out_reason'] = vals.get('opt_out', record.opt_out) and vals.get('opt_out_reason')
+            vals['opt_out_custom'] = vals.get('opt_out', record.opt_out) and vals.get('opt_out_custom')
+            author_id = self.env.ref('base.partner_root').id if record._context.get('from_web_page', False) else self.env.user.partner_id.id
+            if 'opt_out' in vals:
+                message = _('The recipient <strong>unsubscribed from %s</strong> mailing list(s)') \
+                        if vals['opt_out'] else _('The recipient <strong>subscribed to %s</strong> mailing list(s)')
+                record.contact_id.sudo().message_post(body=_(message % record.list_id.name), author_id=author_id)
+                vals['unsubscription_date'] = vals['opt_out'] and fields.Datetime.now()
+
+            if vals['opt_out_reason']:
+                if vals['opt_out_reason'] == 'other':
+                    record.contact_id.sudo().message_post(body=_("Feedback from %s: %s (other)" % (record.contact_id.email, vals['opt_out_custom'])), author_id=author_id)
+                else:
+                    record.contact_id.sudo().message_post(body=_("Feedback from %s: %s" % (record.contact_id.email, dict(record._fields['opt_out_reason']._description_selection(record.env)).get(vals['opt_out_reason']))), author_id=author_id)
```

Comment:
Two messages will be written only if opt_out and opt_out_reason are changed manually at the same time in the backend. Unsubscription and feedback are two distinct actions and they will each trigger a write on the subscription model. Each write will send an unique message. It would be strange to have sometimes both the unsubscription message and the feedback message in the same message and sometimes in two distint message.